src/Microsoft.Security.Utilities.Core/CompiledHighPerformancePattern.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Security.Utilities;
+
+/// <summary>
+/// A pattern that <see cref="HighPerformanceScanner"/> can scan for.
+/// </summary>
+/// <remarks>
+/// There is one instance per signature of patterns that implement <see
+/// cref="IHighPerformanceScannableKey"/>, which is pre-computed using code
+/// generation. Apart from unit tests, no other instances are created.
+/// </remarks>
+internal sealed partial class CompiledHighPerformancePattern
+{
+
+
+    /// <summary>
+    /// The signature that identifies this pattern. This data must be found in
+    /// the input using a fast, initial search operation; otherwise the regex
+    /// will not be applied.
+    /// </summary>
+    public string Signature { get; }
+
+    /// <summary>
+    /// The fixed number of characters that appear before the signature in a match.
+    /// </summary>
+    public int SignaturePrefixLength { get; }
+
+    /// <summary>
+    /// The minimum number of characters that can appear in a match.
+    /// </summary>
+    public int MinMatchLength { get; }
+
+    /// <summary>
+    /// The maximum number of characters that can appear in a match.
+    /// </summary>
+    public int MaxMatchLength { get; }
+
+    /// <summary>
+    /// The regular expression that matches the pattern near a signature.
+    /// <summary>
+    /// <remarks>
+    /// WARNING: This is distinct from the general regexes for this pattern that
+    /// can be applied to find all matches in an arbitrary input. This regex is
+    /// not run on the entire input but only on a substring determined by <see
+    /// cref="SignaturePrefixLength"/> and <see cref="MaxMatchLength"/>.
+    ///
+    /// Example: Let S be a signature character, X be a character--xpected to
```

Comment:
>xpected [](http://example.com/codeflow?start=67&length=7)

` expected`? NIT

Reply:
Sigh, I didn't even finish the example. I'm just going to delete this paragraph for now.